graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphQLQueryRequestTest.kt

```diff
@@ -131,3 +151,31 @@ class MovieProjection : BaseProjectionNode() {
         return this
     }
 }
+
+class DateRangeScalar : Coercing<DateRange, String> {
+    private var formatter = DateTimeFormatter.ofPattern("MM/dd/yyyy")
+
+    @Throws(CoercingSerializeException::class)
+    override fun serialize(dataFetcherResult: Any): String {
+        val range: DateRange = dataFetcherResult as DateRange
+        return range.from.format(formatter) + "-" + range.to.format(formatter)
+    }
+
+    @Throws(CoercingParseValueException::class)
+    override fun parseValue(input: Any): DateRange {
+        val split = (input as String).split("-").toTypedArray()
```

Comment:
In current context it seems only serialize() method is in use. It seems additional overheads to define deserialization methods as its not in use.

Reply:
Is it worth to compose Coercing out of 2 different interfaces (serialization and deserialization), and force client to implement only the serialization part?

Reply:
The `Coercing` type is what all scalars are implemented with, so the goal is to just reuse that code "as is". It's unlikely a user would implement such a scalar _just_ for the client use case I think.

Reply:
:) We have use case that we only consume graphQL API to enrich the data set. But its no deal breaker as i can leave it unimplemented.